scripts/sequence_labeling/data.py

```diff
@@ -337,19 +358,32 @@ def convert_arrays_to_text(text_vocab, tag_vocab,
     for sample_index in range(np_valid_length.shape[0]):
         sample_len = np_valid_length[sample_index]
         entries = []
+        tmptext = ""
         for i in range(1, sample_len - 1):
             token_text = text_vocab.idx_to_token[np_text_ids[sample_index, i]]
             true_tag = tag_vocab.idx_to_token[int(np_true_tags[sample_index, i])]
             pred_tag = tag_vocab.idx_to_token[int(np_pred_tags[sample_index, i])]
             # we don't need to predict on NULL tags
-            if true_tag == NULL_TAG:
-                last_entry = entries[-1]
-                entries[-1] = PredictedToken(text=last_entry.text + token_text,
-                                             true_tag=last_entry.true_tag,
-                                             pred_tag=last_entry.pred_tag)
+            if tagging_first_token:
+                if true_tag == NULL_TAG:
+                    last_entry = entries[-1]
+                    entries[-1] = PredictedToken(text=last_entry.text + token_text,
+                                                 true_tag=last_entry.true_tag,
+                                                 pred_tag=last_entry.pred_tag)
+                else:
+                    entries.append(PredictedToken(text=token_text,
+                                                  true_tag=true_tag, pred_tag=pred_tag))
             else:
-                entries.append(PredictedToken(text=token_text,
-                                              true_tag=true_tag, pred_tag=pred_tag))
-
+                if true_tag == NULL_TAG:
+                    tmptext += token_text
+                else:
+                    if len(tmptext) > 0:
+                        text = tmptext + token_text
+                        entries.append(PredictedToken(text=text,
+                                                      true_tag=true_tag, pred_tag=pred_tag))
+                        tmptext = ''
+                    else:
```

Comment:
Can both cases be merged here? For example, if `len(tmptext) == 0`, you can still have `text = tmptext + token_text` which is equivalent to `token_text`.